runtime/util/hshelp.c

```diff
@@ -3046,6 +3046,33 @@ verifyClassesAreCompatible(J9VMThread * currentThread, jint class_count, J9JVMTI
 			return rc;
 		}
 
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+		/* Verify that the nestmates attributes - nest host & nest members - are the same */
+
+		if (J9ROMCLASS_NESTHOSTNAME(originalROMClass) != J9ROMCLASS_NESTHOSTNAME(replacementROMClass)) {
```

Comment:
Should this be a UTF comparison rather than identity?

Reply:
I suppose some NULL checking would need to be done as well.

Reply:
Missed that - thanks

Reply:
Missed that - thanks. Corrected now.